Src/WindowsServer/WindowsServer.Shared/Implementation/AzureRoleEnvironmentContextReader.cs

```diff
@@ -62,25 +65,97 @@ internal set
         /// </summary>
         internal static string BaseDirectory { get; set; }
 
+        internal static string AssemblyName { get; set; }
+
+        internal static string Culture { get; set; }
+
+        internal static string PublicKeyToken { get; set; }
+
+        internal static string[] VersionsToAttempt { get; set; }
+
         /// <summary>
         /// Initializes the current reader with respect to its environment.
         /// </summary>
         public void Initialize()
         {
-            ServiceRuntime serviceRuntime = new ServiceRuntime();
-            RoleEnvironment roleEnvironment = serviceRuntime.GetRoleEnvironment(AzureRoleEnvironmentContextReader.BaseDirectory);
+            AppDomain tempDomainToLoadAssembly = null;
+            string tempDomainName = "AppInsightsDomain-" + Guid.NewGuid().ToString().Substring(0, 6);
```

Comment:
forgot to ask - why `substring(0,6)`? Did you hit some limitations in appdomain name length?